backend/src/queues/secret-scanning/githubScanFullRepository.ts

```diff
@@ -22,26 +24,77 @@ type TScanPushEventQueueDetails = {
 
 githubFullRepositorySecretScan.process(async (job: Job, done: Queue.DoneCallback) => {
   const { organizationId, repository, installationId }: TScanPushEventQueueDetails = job.data
+  const [owner, repo] = repository.fullName.split("/");
+
   try {
     const octokit = new ProbotOctokit({
         auth: {
         appId: await getSecretScanningGitAppId(),
         privateKey: await getSecretScanningPrivateKey(),
         installationId: installationId
-        },
+      },
     });
+
+    // Scan the .infisicalignore file (if it exists) & extract the fingerprints
+    const infisicalIgnoreFileContents = await checkIfInfisicalIgnoreFile(octokit, owner, repo);
+    const batchUpdateOperations: any[] = [];
     const findings : SecretMatch[] = await scanFullRepoContentAndGetFindings(octokit, installationId, repository.fullName)
+
     for (const finding of findings) {
-      await GitRisks.findOneAndUpdate({ fingerprint: finding.Fingerprint}, 
-        {
-        ...convertKeysToLowercase(finding),
-        installationId: installationId,
-        organization: organizationId,
-        repositoryFullName: repository.fullName,
-        repositoryId: repository.id
-      }, {
-        upsert: true
-      }).lean()
+
+      // Create a SHA3-512 hash of the secret & immediately clear the plaintext secret from memory
+      // Used in push events to collate all SHA-3 hashes of the secret in the same repo's file (prevents case where the secret moves)
+      // Also useful to append to the fingerprint (eventually) & for user audit secret scanning log table .csv download
+      let secret = finding.Secret;
+      const sha512Hash = createHash("sha3-512");
+      sha512Hash.update(secret);
+      const hashResult = sha512Hash.digest("hex");
+      secret = "";
+      
+      const updateOperation = {
+        updateOne: {
+          filter: { fingerprint: finding.Fingerprint },
+          update: {
+            ...convertKeysToLowercase(finding),
+            installationId: installationId,
+            organization: organizationId,
+            repositoryFullName: repository.fullName,
+            repositoryId: repository.id,
+            hashedSecret: hashResult,
```

Comment:
@dangtony98 do you have some thoughts on this? I've used SHA3-512 here. The idea of putting the secret hash in is to more efficiently look through the repo and eventually search other repos for that secret. GitGuardian for example returns a 64 char base64 string ("secret_hash") when you download their .csv table. It's currently not exposed on frontend but maybe with RBAC an admin could get the table with the hashes?

Reply:
I think we can use Blind indexing here so that the secret is encrypted, but we can still query for it. We use blind indexing to encrypt secret names and then query for them later https://github.com/Infisical/infisical/blob/c3696bdbbc06e5bc13c78cc9fe19d321d6092f65/backend/src/services/SecretService.ts#L82

@dangtony98 can you confirm?

Reply:
Spoke to Tony, he also agrees that blind indexing should be added for this. You will likely have to create another table in the db to store these blind index details (we have one for secrets). Feel free to ask Tony about the specifics if you need help